core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -169,12 +169,13 @@ private[deploy] object DependencyUtils extends Logging {
   }
 
   /**
-   * Merge a sequence of comma-separated file lists, some of which may be null to indicate
-   * no files, into a single comma-separated string.
+   * Merge and de-duplicate a sequence of comma-separated file lists,
+   * some of which may be null to indicate no files,
+   * into a single comma-separated string.
    */
   def mergeFileLists(lists: String*): String = {
     val merged = lists.filterNot(StringUtils.isBlank)
-      .flatMap(Utils.stringToSeq)
+      .flatMap(Utils.stringToSeq).distinct
```

Comment:
Sorry, I have a few questions for you that might be trivial but I'm unclear about them. I was wondering why do we need the `.distinct`? Obviously, I know that it removes the duplicated values, but how does it change how the function is used/what implications does it have?

Thank you!

Reply:
Avoid uploading the same file

https://github.com/apache/spark/blob/5c7d019b609c87a9427fa9309f3aa03d02f61878/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala#L455-L460